day1/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+const INITIAL_FLOOR: i32 = 0;
+
+/// Updates the floor Santa would reach after following the passed `instruction`.
+/// Returns `Some(new_floor)` to be compatible with [`std::iter::Iterator::scan`].
+fn process_floor(current_floor: &mut i32, instruction: char) -> Option<i32> {
+    match instruction {
+        '(' => *current_floor += 1,
+        ')' => *current_floor -= 1,
+        _ => panic!("invalid input"),
+    }
+    Some(*current_floor)
+}
+
+/// Returns the last floor Santa would be on after following all the instructions in `input`.
+pub fn part1(input: &str) -> i32 {
+    input
+        .chars()
+        .scan(INITIAL_FLOOR, process_floor)
+        .last()
+        .unwrap()
+}
+
+/// Returns the first day Santa be enter the basement on while following all the instructions in
+/// `input`.
+pub fn part2(input: &str) -> usize {
+    input
+        .chars()
+        .scan(INITIAL_FLOOR, process_floor)
+        .zip(1..)
+        .find(|&(floor, _)| floor < 0)
+        .unwrap()
+        .1
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_part_1_examples() {
+        assert_eq!(part1("(())"), 0);
+        assert_eq!(part1("()()"), 0);
+        assert_eq!(part1("((("), 3);
+        assert_eq!(part1("(()(()("), 3);
+        assert_eq!(part1("))((((("), 3);
+        assert_eq!(part1("())"), -1);
+        assert_eq!(part1("))("), -1);
+        assert_eq!(part1(")))"), -3);
+        assert_eq!(part1(")())())"), -3);
+    }
+
+    #[test]
+    fn test_part_2_examples() {
+        assert_eq!(part2(")"), 1);
+        assert_eq!(part2("()())"), 5);
+    }
+
+    #[test]
+    fn test_part_1_chris_input() {
+        let root = std::env::var("CARGO_MANIFEST_DIR").unwrap();
```

Comment:
These seem like a good spot for `include_str!` instead.

Reply:
I'd love to see it!

Completely agreed since
* the file contents are known at compile time
* it would be nice to avoid the `std::env::var("CARGO_MANIFEST_DIR")` stuff